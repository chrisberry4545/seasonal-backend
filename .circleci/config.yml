version: 2
jobs:
  test:
    docker:
      - image: circleci/node:8.10

    steps:
      - checkout

      - setup_remote_docker

      - run:
          name: Output dir
          command: echo $PWD
      - run:
          name: Run Docker
          command: docker-compose run app npm test

      - run:
          name: Run tests
          command: docker exec app npm run test

  deploy:
    docker:
      - image: circleci/node:8.10

    steps:
      - checkout

      - restore_cache:
          keys:
          - v1-dependencies-
          # fallback to using the latest cache if no exact match is found
          - v1-dependencies-

      - run:
          name: Copy env config
          command: |
            cp .env-ci.yml .env.yml

      - run:
          name: Install Serverless CLI and dependencies
          command: |
            sudo npm i -g serverless
            npm install

      - run:
          name: Run tests
          command: npm test

      - run:
          name: Build dist version
          command: |
            npm run build
            npm run move-dist-to-top-level

      - save_cache:
          paths:
            - node_modules
          key: v1-dependencies-

      - run:
          name: Deploy application
          command: sls deploy --stage prod

workflows:
  version: 2
  ci:
    jobs:
      - deploy:
          filters:
            branches:
              only:
                - master
      - test
