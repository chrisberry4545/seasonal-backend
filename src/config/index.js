module.exports = {
  AIRTABLE_API_KEY: process.env.AIRTABLE_API_KEY,
  SKIP_CACHE: process.env.SKIP_CACHE || false
};
