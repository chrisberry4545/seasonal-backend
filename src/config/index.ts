export const SKIP_CACHE = process.env.SKIP_CACHE || false;
export const SEASON_ENDPOINT =
  process.env.SEASON_ENDPOINT || 'season';
export const FOOD_ENDPOINT =
  process.env.FOOD_ENDPOINT || 'food';
export const SEASON_WITH_FOOD_ENDPOINT =
  process.env.SEASON_WITH_FOOD_ENDPOINT || 'season-with-food';
export const SEASON_WITH_RECIPES_ENDPOINT =
  process.env.SEASON_WITH_RECIPES_ENDPOINT || 'season-with-recipes';
export const COUNTRY_ENDPOINT =
  process.env.COUNTRY_ENDPOINT || 'country';
export const V2_ENDPOINT = process.env.V2_ENDPOINT || 'v2';
export const DB_CONNECTION_STRING = process.env.DB_CONNECTION_STRING;

export const DEFAULT_COUNTRY_ID =
  process.env.DEFAULT_COUNTRY_ID || '83533319-9b7a-4c79-9261-4c02bc6b4d0d';
