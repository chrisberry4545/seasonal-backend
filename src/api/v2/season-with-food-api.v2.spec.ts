import {
  app
} from '../../app';

import supertest, { Response } from 'supertest';
import { V2_ENDPOINT, SEASON_WITH_FOOD_ENDPOINT } from '../../config';

const v2SeasonWithFoodUrl = `${V2_ENDPOINT}/${SEASON_WITH_FOOD_ENDPOINT}`;

describe('Get all seasons with food', () => {
  let response: Response;
  beforeAll(async () => {
    response = await supertest(app).get(`/${v2SeasonWithFoodUrl}`);
  });

  test('Returns a status of 200', () => {
    expect(response.status).toBe(200);
  });
  test('Returns a full list of season data', () => {
    expect(response.body).toMatchSnapshot();
  });
  test('Returns the name of the first season', () => {
    expect(response.body[0].name).toBe('January');
  });
  test('Returns the name of the second season', () => {
    expect(response.body[1].name).toBe('February');
  });
  test('Populates the food in the seasons', () => {
    expect(response.body[0].food).toHaveLength(1);
  });
  test('Populates the food names in the seasons', () => {
    expect(response.body[0].food[0].name).toBe('Beetroot');
  });
});

describe('Get single season with food', () => {
  let response: Response;
  const makeSingleSeasonWithFoodRequest = (id: string = '0') => {
    return supertest(app).get(`/${v2SeasonWithFoodUrl}/${id}`);
  };

  describe('when the requested season has no food', () => {
    beforeAll(async () => {
      response = await makeSingleSeasonWithFoodRequest();
    });

    test('Returns a status of 200', () => {
      expect(response.status).toBe(200);
    });
    test('Retrieves a single season', () => {
      expect(response.body).toMatchSnapshot();
    });
    test('Does not populate a seasons food if they do not exist', () => {
      expect(response.body.food).toBeUndefined();
    });
  });

  describe('when the requested season has food data', () => {
    beforeAll(async () => {
      response = await makeSingleSeasonWithFoodRequest('1');
    });

    test('Retrieves a single season with food data', () => {
      expect(response.body).toMatchSnapshot();
    });
    test('Populates a seasons food if they exist', () => {
      expect(response.body.food.length > 0).toBe(true);
    });
    test('Does not return any recipes', () => {
      expect(response.body.recipes).toBeUndefined();
    });
  });
});
