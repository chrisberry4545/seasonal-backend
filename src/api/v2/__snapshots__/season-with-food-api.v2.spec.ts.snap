// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Get all seasons with food Returns a full list of season data 1`] = `
Array [
  Object {
    "food": Array [
      Object {
        "id": "dd9ba012-8f8e-48af-9775-0139374dd94c",
        "imageUrlSmall": "https://apple.com",
        "name": "Apple",
      },
      Object {
        "id": "c6f78568-fe23-47e4-8e65-55934199a39f",
        "imageUrlSmall": "https://beetroot.com",
        "name": "Beetroot",
      },
    ],
    "id": "41bcbb3a-5599-4f0c-8da9-3144cc5be6de",
    "name": "January",
    "seasonIndex": "0",
  },
  Object {
    "food": Array [
      Object {
        "id": "c6f78568-fe23-47e4-8e65-55934199a39f",
        "imageUrlSmall": "https://beetroot.com",
        "name": "Beetroot",
      },
    ],
    "id": "eec0b2bb-4fbd-46df-b905-8d2ee2fb991a",
    "name": "February",
    "seasonIndex": "1",
  },
  Object {
    "food": Array [],
    "id": "f4812071-b0e2-4255-8ccf-e38768ad0c3c",
    "name": "March",
    "seasonIndex": "2",
  },
]
`;

exports[`Get single season with food when the requested season has food data Retrieves a single season with food data 1`] = `
Object {
  "food": Array [
    Object {
      "id": "c6f78568-fe23-47e4-8e65-55934199a39f",
      "imageUrlSmall": "https://beetroot.com",
      "name": "Down Under Beetroot",
    },
  ],
  "id": "eec0b2bb-4fbd-46df-b905-8d2ee2fb991a",
  "name": "February",
  "seasonIndex": "1",
}
`;

exports[`Get single season with food when the requested season has no food Retrieves a single season 1`] = `
Object {
  "food": Array [],
  "id": "f4812071-b0e2-4255-8ccf-e38768ad0c3c",
  "name": "March",
  "seasonIndex": "2",
}
`;
